### PR TITLE
docs: explain IndieAuth

### PR DESCRIPTION
**Pull request by sebsel**
Some new docs about what Selfauth is.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
-# Warning: while Selfauth will work with old versions of PHP, the ability to get sufficiently random strings was not added until version 5.6.  While older versions are not completely insecure, it is strongly recommended you upgrade to a newer version of PHP.
+Selfauth is a self-hosted [Authorization Endpoint](https://indieweb.org/authorization-endpoint) used to login with a personal URL (as [Web sign-in](http://indieweb.org/Web_sign-in)) via [IndieAuth](https://indieweb.org/IndieAuth). See [How it works](#how-it-works) for more.
 
-# Warning: currently selfauth only supports authentication, not authorization.  Meaning scopes will not work yet.  If you need access to these features, it is advisable to use something else for now.
 
-Setup
------
+## Warnings
+
+- While Selfauth will work with old versions of PHP, the ability to get sufficiently random strings was not added until version 5.6. While older versions are not completely insecure, **it is strongly recommended you upgrade to a newer version of PHP**.
```
Comment (Zegnat) [inline README.md]: It isn’t just random strings anymore, also things like `hash_equals`. And possibly the `password_`-functions if we end up using those.

Can we have a more general description that for PHP versions below 5.6 we can’t use PHP’s security features and that we cannot guarantee the same full security even though Selfauth will still be functional?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
+
+- Currently selfauth only supports authentication, not authorization.  Meaning **scopes will not work yet**. If you need access to these features, it is advisable to use something else for now.
```
Comment (Zegnat) [inline README.md]: This isn’t strictly true. Selfauth will happily issue a code for authorisation use, with scopes included in the HMAC and code.

File: README.md (hunk 1 above)
Comment (sebsel) [reply README.md]: I made this "While Selfauth will work with old versions of PHP, some of the more secure functions Selfauth uses were not added until version 5.6." on my local copy now.

File: README.md (hunk 2 above)
Comment (sebsel) [reply README.md]: At the moment, this is true. I created #19, and once that closes we can remove / change this warning.

File: README.md [hunk 3]
```diff
@@ -21,13 +25,22 @@ To set up Selfauth, create a folder on your webserver and add the files in this
...
+
+To change your password, make sure the `setup.php` file is in place again and delete `config.php`. Then follow the steps under [Setup](#setup) again.
+
+
+## How it works
+
+On a (Web)App which supports [IndieAuth](https://indieweb.org/IndieAuth), you can enter your personal URL. The App will detect Selfauth as Authorization Endpoint and redirect you to it. After you enter your password in Selfauth, you are redirected back to the App with a code. The App will verify the code with Selfauth and logs you in as your personal URL.
 
-5. Fill in your password and click login!
+To test it, you can go to an App that supports IndieAuth and enter your personal URL. [IndieAuth.com](https://indieauth.com/) has a test-form on the frontpage. If you also link to your social media accounts using `rel="me"`, IndieAuth.com might show you a list of buttons, on which you can click the one that has your Selfauth URL on it.
```
Comment (dissolve) [inline README.md]: Last sentence here is really confusing.  Suggest splitting it to two sentences. 

IndieAuth.com might show you a list of buttons.  To use selfauth, click on the one that ...

**Review by dissolve — CHANGES_REQUESTED** [review]
One text change, the rest looks good to me

**Comment by dissolve** [comment]
Actually, this mentions that setup.php stops working, is that true yet?

**Comment by Zegnat** [comment]
> Actually, this mentions that setup.php stops working, is that true yet?

I think this refers to the fact that setup.php will not create/overwrite config.php if it already exists.

**Comment by dissolve** [comment]
its also moot as there is a PR for that now anyway :smile:

**Review by dissolve — APPROVED** [review]

**Pull request merged**